worlds/oot_soh/__init__.py

```diff
@@ -241,6 +241,7 @@ def fill_slot_data(self) -> dict[str, Any]:
             "ice_trap_count": self.options.ice_trap_count.value,
             "ice_trap_filler_replacement": self.options.ice_trap_filler_replacement.value,
             "apworld_version": self.apworld_version,
+            "enable_all_tricks": self.options.enable_all_tricks.value,
             "tricks_in_logic": self.options.tricks_in_logic.value
```

Comment:
I feel like instead of passing `enable_all_tricks` into slot data, it should instead just modify the `tricks_in_logic` option accordingly during generate_early.